### ISSUE TITLE
add pronoun.is/none for by-name-only preference

### ISSUE THREAD

**Issue by jraqula** [witch-house/pronoun.is#45]
It would be great to have an option for preferring no pronouns, i.e. "James threw the frisbee/I think it is James's/etc". This might be a bit complicated based on how the database is pulled from, so I figured I'd raise a suggestion rather than a pull request. I suggest www.pronoun.is/none as the route. 💃

**Comment by morganastra** [comment]
If the route was at pronoun.is/none, how would we know what name to use in the examples?

What would the reflexive be? "Name's self"?

just to make sure I understand you correctly, you're imagining it displaying something equivalent to http://pronoun.is/James/James/James's/James's/James's%20self ?

**Comment by jraqula** [comment]
Yeah, "___ self" is the paradigmatic solution i've come up with, though in my experience the reflexive is so rarely an issue when speaking. So, that route you gave would have equivalent output, if we used the name James.

The question of which name to use is a good one, and part of why I suspected this is a bit complicated. I suppose if you created a singleton case for it it would be easy to provide custom behavior/a default name?

Since the only name that appears on the site is yours, it might be sensible to use Morgan as a name. However that might confuse folks since your pronoun.is/ze/zir ! Any thoughts?

**Comment by morganastra** [comment]
okay now that we use query parameters for something (multi pronouns, see #38), we can use it everywhere! what do you think about having an url of the form `pronoun.is/?name=James?` (I'm definitely open to suggestions for what that parameter should be called to make the url flow nicely when spoken aloud)

**Comment by ardnaxelarak** [comment]
I like the idea of combining the two suggestions, so that it would be `pronoun.is/none?name=James` since that sounds like "my pronouns are none, my name is James" spoken aloud. This has the drawback of having to make none a special case (could map it to a separate route if need be), but this doesn't look any worse than the special case of `pronoun.is/` returning the info page with no param and the pronoun list with a `name` param.

**Comment by TazeTSchnitzel** [comment]
I've heard people phrase it as “proper nouns”. So, `pronoun.is/proper-noun`?

**Comment by morganastra** [comment]
I don't have a strong preference between `/none` and `/proper-noun`; input from someone who actually doesn't use pronouns would be the correct way to decide. (for clarity, I had assumed that was @jraqula, but I see that their pronoun is listed as they or she)

Regardless of the route, there are some other outstanding issues, including but probably not limited to:
  * the language explaining what the site is would be wrong for the no-pronoun-just-name case
  * names are not personal pronouns
  * "Here are some usage examples for my Names/Names pronouns" is a silly and misleading/inaccurate title

So at the least, we'd have to write a whole new page for this. This probably won't happen for a while.

**Comment by jraqula** [comment]
Hey y'all, sorry i've been out of touch. I noted they/them basically because these explanations were too expensive for the "proper noun" route which is still hiding in my ❤️ of 💕. I think after reviewing all that's been said above @TazeTSchnitzel 's suggestion is most appealing to me, could even have just a brief paragraph description of what it means. 

To your first outstanding issue, you're referring to "to show people how to use pronouns in English"? For what it's worth, I wouldn't be disturbed by finding a page about the deliberate avoidance of pronoun usage on a sight about pronoun usage, given that non-usage is a decision in the same scope. Also, I doubt any amount of rhetorically intentful apositionality could liberate the site from the context of gender identity, and it would be in keeping with that purpose to have a page like this.

That being said, it totally makes sense to deprioritize this, especially if it is a substantial deviation from the patterns currently in place. If we came to a point of having a design spec we were happy with, I could add it to my to-do list however.

**Comment by bmintz** [comment]
Although I can't speak for those that don't use gender pronouns, I would _personally_ prefer `/none` simply because`my.pronoun.is/none` reads  closer to "I don't use gender pronouns" than `my.pronoun.is/proper-noun`, since proper nouns aren't pronouns.

Or we could just support both endpoints, since there seems to be people on both sides of this issue.

Also, +1 to a `name` parameter.

**Comment by morganastra** [comment]
Marking this as "help wanted". I think the arguments for the `/none` endpoint are convincing. I would accept a PR for this which:
    * Added a route for `/none` with a `name` parameter. I'm still not sure what it should do in the case where no `name` is given, I'll want whoever takes this on to have a proposal for that.
    * Added the `/none` page with some language explaining the concept of not using pronouns and a title and header text that make sense.
    * Please don't do this using a hack on the pronouns database; "none" isn't a pronoun.

**Comment by bmintz** [comment]
Feel free to assign this to me.

This is my first time with Clojure, though, so it might take me a while.

**Comment by Elizafox** [comment]
Hi,

I prefer not to use pronouns. But she/they are also acceptable. So... whatever is done should accept supplementary pronouns, like pronoun.is/none?name=Eliza&or=she&or=they

I could see some people having a preference for a pronoun but falling back to none, but this case seems nonexistent in practise.

If a name is omitted, a simple explanation of what using no pronoun means.

For me, I don't use a reflexive. It just doesn't sound right.

Something simple like "This person does not use/does not prefer pronouns and prefers their proper name." as the description is fine, with a few examples.

This is all mighty subjective but that's my two cents.

**Comment by morganastra** [comment]
@bmintz if you're still an interested in this, please feel free to pick it up!

I think I'm settling on the convention of using `:` in front of path elements that aren't pronouns just to completely avoid the chance of confusion with the pronoun database, so the url should look like `pronoun.is/:none?name=Whatever`. 

In the case where no `name` parameter is provided, just use the string "Name" as a substitute, unless you can think of a better one.

Supporting none + other pronoun examples is harder so let's not do that for now.

Let me know if you have any other questions!

(If anybody else wants to pick this up you're welcome to also, just comment here so there's not two people working on it at once)

**Comment by bmintz** [comment]
@morganastra how about Riley, a common first name in the united states which is almost split between sexes? https://fivethirtyeight.com/features/there-are-922-unisex-names-in-america-is-yours-one-of-them/

**Comment by bmintz** [comment]
And yes, I will take it on. I think there's a way inside GitHub to actually assign it to me, but if you don't want to do that that's fine too.
